lib/src/value/calculation.dart

```diff
@@ -158,6 +343,240 @@ class SassCalculation extends Value {
+    var args = [dividend, if (modulus != null) modulus];
+    _verifyLength(args, 2);
+    _verifyCompatibleNumbers(args);
+    if (dividend is! SassNumber ||
+        modulus is! SassNumber ||
+        !dividend.hasCompatibleUnits(modulus)) {
+      return SassCalculation._("mod", args);
+    }
+    return dividend.modulo(modulus);
+  }
+
+  /// Creates a `round()` calculation with the given [strategyOrNumber], [numberOrStep], and [step].
+  /// Strategy must be either nearest, up, down or to-zero.
+  ///
+  /// Number and step must be either a [SassNumber], a [SassCalculation], an
+  /// unquoted [SassString], a [CalculationOperation], or a
+  /// [CalculationInterpolation].
+  ///
+  /// This automatically simplifies the calculation, so it may return a
+  /// [SassNumber] rather than a [SassCalculation]. It throws an exception if it
+  /// can determine that the calculation will definitely produce invalid CSS.
+  ///
+  /// This may be passed fewer than two arguments, but only if one of the
+  /// arguments is an unquoted `var()` string.
+  static Value round(Object strategyOrNumber,
+      [Object? numberOrStep, Object? step]) {
+    switch ((
+      _simplify(strategyOrNumber),
+      numberOrStep.andThen(_simplify),
+      step.andThen(_simplify)
+    )) {
+      case (SassNumber number, null, null):
+        return _matchUnits(number.value.round().toDouble(), number);
+
+      case (SassNumber number, SassNumber step, null):
+        _verifyCompatibleNumbers([number, step]);
+        return _roundWithStep('nearest', number, step);
+
+      case (
+          SassString(text: 'nearest' || 'up' || 'down' || 'to-zero') &&
+              var strategy,
+          SassNumber number,
+          SassNumber step
+        ):
+        _verifyCompatibleNumbers([number, step]);
+        return _roundWithStep(strategy.text, number, step);
+
+      case (
+          SassString(text: 'nearest' || 'up' || 'down' || 'to-zero') &&
+              var strategy,
+          (SassString() || CalculationInterpolation()) && var rest?,
+          null
+        ):
+        return SassCalculation._("round", [strategy, rest]);
+
+      case (
+          SassString(text: 'nearest' || 'up' || 'down' || 'to-zero'),
+          _?,
+          null
+        ):
+        throw SassScriptException("If strategy is not null, step is required.");
+
+      case (
+          SassString(text: 'nearest' || 'up' || 'down' || 'to-zero'),
+          null,
+          null
+        ):
+        throw SassScriptException(
+            "Number to round and step arguments are required.");
+
+      case (
+          (SassString() || CalculationInterpolation()) && var strategy,
+          null,
+          null
+        ):
+        return SassCalculation._("round", [strategy]);
+
+      case (var number, null, null):
+        throw SassScriptException(
+            "Single argument $number expected to be simplifiable.");
+
+      case (var strategy, var number?, null):
```

Comment:
Nit: these should be `number` and `step`, since that's what the two-argument plain-CSS `round()` function takes.